config/nuttx/stm32f4dis/patches/0004-Provide-working-iotjs-config-for-7.19-with.patch

```diff
@@ -0,0 +1,1818 @@
+From d99da54fd7296a7ac02bed17a7ded0f6096d283c Mon Sep 17 00:00:00 2001
+From: Tomasz Wozniak <t.wozniak@samsung.com>
+Date: Wed, 23 Aug 2017 12:45:48 +0200
+Subject: [PATCH 4/4] Provide working iotjs config for 7.19 with:  -FPU/MPU 
+ -ROMFS, TMPFS  -USB/NSH  -ADC, I2C, SPI, UART, GPIO, TCP/IP, UDP
+
+---
+ configs/stm32f4discovery/usbnshiotjs/Make.defs |  113 ++
+ configs/stm32f4discovery/usbnshiotjs/defconfig | 1589 ++++++++++++++++++++++++
+ configs/stm32f4discovery/usbnshiotjs/setenv.sh |   80 ++
+ 3 files changed, 1782 insertions(+)
+ create mode 100644 configs/stm32f4discovery/usbnshiotjs/Make.defs
+ create mode 100644 configs/stm32f4discovery/usbnshiotjs/defconfig
+ create mode 100755 configs/stm32f4discovery/usbnshiotjs/setenv.sh
```

Comment:
I prefer a patch, because the contents, location and validation are kept in a single place.
With file copy, one needs to know:
 - where to copy the file,
 - should there be no file, or overwrite is okay?
And we lose all the validation that patch applying does (I prefer to know the patch does not apply, than just blindly overwrite a file that might have changed).

I would also like to replace our current apps/system/iotjs directory copy with a patch. Maybe later, definitely not in this patch.